src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImplTest.java

```diff
@@ -1,7 +1,67 @@
 package pl.byczazagroda.trackexpensesappbackend.service;
 
-import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.FilterType;
+import pl.byczazagroda.trackexpensesappbackend.controller.WalletController;
+import pl.byczazagroda.trackexpensesappbackend.dto.UpdateWalletDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.WalletDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.WalletModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.exception.WalletNotFoundException;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
 
+import java.time.Instant;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.BDDMockito.given;
+
+@WebMvcTest
+        (controllers = WalletController.class,
+                includeFilters = @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, classes = {
+                        WalletRepository.class,
+                        WalletServiceImpl.class}))
 class WalletServiceImplTest {
 
+    @MockBean
+    private WalletRepository walletRepository;
+
+    @MockBean
+    private WalletModelMapper mapper;
+
+    @Autowired
+    private WalletServiceImpl underTest;
```

Comment:
We can change the type to WalletService in line with dependency inversion principle